### PR TITLE
rdl_schema_0_1.json: minor updates to field names, titles and descriptions, add risk_data_type codelist and documentation

### PR DESCRIPTION
**Pull request by odscjen**
Some of the minor updates listed in #94, primarily updates to field names, titles and descriptions in the top level objects that aren't involved in any other issues.

Also updated `definitions` to `$defs`

### CODE REVIEW

File: schema/rdl_schema_0.1.json [hunk 1]
```diff
@@ -17,16 +16,19 @@
...
 			"type": "string",
-			"description": "Short description of the resource"
+			"description": "A short description of the dataset."
 		},
-		"component": {
+		"risk_data_type": {
+			"title": "Risk data type",
 			"type": "string",
-			"description": "Univocal schema for this contribution (H, E, V or L)",
+			"description": "The datasets risk data type.",
```
Comment (duncandewhurst) [inline schema/rdl_schema_0.1.json:31]: Need to add a CSV file for this codelist and add the codelist keywords.

```suggestion
			"description": "The type of risk data included in the dataset, from the closed risk data types codelist.",
```

File: schema/rdl_schema_0.1.json [hunk 2]
```diff
@@ -48,20 +50,24 @@
 			"format": "date"
 		},
 		"version": {
+			"title": "Dataset version",
 			"type": "integer",
-			"description": "Version of the dataset"
+			"description": "The version of the dataset."
```
Comment (duncandewhurst) [inline schema/rdl_schema_0.1.json:55]: Suggest aligning with DCAT's [`version`](https://www.w3.org/TR/vocab-dcat-3/#Property:resource_version):
```suggestion
			"description": "The version indicator (name or identifier) of the dataset."
```

File: schema/rdl_schema_0.1.json [hunk 3]
```diff
@@ -48,20 +50,24 @@
...
 		},
 		"purpose": {
+			"title": "Dataset purpose",
 			"type": "string",
-			"description": "Purpose for what the data has been produced"
+			"description": "The purpose the dataset was produced for."
```
Comment (duncandewhurst) [inline schema/rdl_schema_0.1.json:60]: Copy edit to use more formal language:

```suggestion
			"description": "The purpose for which the dataset was created."
```

File: schema/rdl_schema_0.1.json [hunk 4]
```diff
@@ -48,20 +50,24 @@
...
 		},
 		"project": {
+			"title": "Project title",
 			"type": "string",
-			"description": "Project under which data has been produced"
+			"description": "The name of the project that the dataset was created as part of."
```
Comment (duncandewhurst) [inline schema/rdl_schema_0.1.json:65]: Copy edit for clarity and brevity:

```suggestion
			"description": "The name of the project that produced the dataset."
```

**Review by duncandewhurst — CHANGES_REQUESTED** [review]

**Comment by odscjen** [comment]
I can't see which csv file is causing the issue

**Comment by duncandewhurst** [comment]
Here's the error from [line 85](https://github.com/GFDRR/rdl-standard/actions/runs/5403490702/jobs/9816371927?pr=100#step:6:86) of the test output:

```
tests/test_csv.py::test_valid
  ERROR: specs/code-lists/CSV/risk_data_type.csv is improperly formatted (e.g. missing trailing newline, extra quoting characters, non-"\n" line terminator):
  'Code,Title,Description\nHazard,Hazard,Hazard risk data\nExposure,Exposure,Exposure risk data\nVulnerability,Vulnerability,Vulnerability risk data\nLoss,Loss,Loss risk data'
  'Code,Title,Description\nHazard,Hazard,Hazard risk data\nExposure,Exposure,Exposure risk data\nVulnerability,Vulnerability,Vulnerability risk data\nLoss,Loss,Loss risk data\n'
```

**Comment by duncandewhurst** [comment]
I added few commits:

* Add an entry to the codelist reference page for the risk_data_type codelist (by running `./manage.py pre-commit`)
* Add a link to the codelist reference from the schema
* Add a stub schema reference page so that the links to referencing properties in the codelist reference documentation are resolvable. We'll integrate the schema reference page with the data model reference documentation in another PR.
* Made some changes to the build process so that those links work on Read the Docs regardless of the branch you're on.

You can see this working here: https://rdl-standard.readthedocs.io/en/94_minor_changes_top_level_fields/data_model/schema/#rdl_schema_0.1.json,,risk_data_type

@stufraser1 don't worry about reviewing the changes to the configuration files. I think that you can focus your review on [`risk_data_type.csv`](https://github.com/GFDRR/rdl-standard/pull/100/files#diff-2f89a06888fdb864d49a649e9931a81bcefdffcacf613700a5066a13d54d8caf) and [`rdl_schema_0.1.json`](https://github.com/GFDRR/rdl-standard/pull/100/files#diff-027a974120cd59c95e322a021f24983ada40ed519431d9d3de52edcde718f480).

**Review by duncandewhurst — APPROVED** [review]

File: schema/rdl_schema_0.1.json [hunk 5]
```diff
@@ -17,16 +16,21 @@
   ],
   "properties": {
     "title": {
+      "title": "Dataset title",
       "type": "string",
-      "description": "Title of the dataset"
+      "description": "The title of the dataset."
```
Comment (stufraser1) [inline schema/rdl_schema_0.1.json:21]: Does "The Title of the dataset" in `description` not duplicate `title`?

**Review by stufraser1 — CHANGES_REQUESTED** [review]
changes committed to risk_data_type.csv, comment left on rdl_schema_0.1.json

File: schema/rdl_schema_0.1.json (hunk 5 above)
Comment (odscjen) [reply schema/rdl_schema_0.1.json:21]: yes, I've adjusted to make them less duplicates

**Review by stufraser1 — APPROVED** [review]

**Pull request merged**